runtime/vm/vmthread.cpp

```diff
@@ -205,6 +205,23 @@ allocateVMThread(J9JavaVM * vm, omrthread_t osThread, UDATA privateFlags, void *
 	newThread->stackObject = stack;
 	newThread->stackOverflowMark = newThread->stackOverflowMark2 = J9JAVASTACK_STACKOVERFLOWMARK(stack);
 	newThread->osThread = osThread;
+#if defined(J9VM_OPT_CRIU_SUPPORT)
+	/* JDWP threads need to remain live while checkpoint/restore hooks run, so add
+	* J9_PRIVATE_FLAGS2_DELAY_HALT_FOR_CHECKPOINT to the new J9VMThread created from a
+	* JDWP java thread object.
+	*/
+	if (J9_ARE_ALL_BITS_SET(vm->checkpointState.flags, J9VM_CRIU_IS_JDWP_ENABLED)) {
+		for (UDATA i = 0; i < vm->checkpointState.javaDebugThreadCount; i++) {
+			if (NULL != vm->checkpointState.javaDebugThreads[i]) {
```

Comment:
Isn't this test redundant? How could it ever be false?

Reply:
You're right, removed the null check.